connect/runtime/src/main/java/org/apache/kafka/connect/runtime/WorkerTask.java

```diff
@@ -291,9 +293,21 @@ protected boolean awaitUnpause() throws InterruptedException {
 
     public void transitionTo(TargetState state) {
         synchronized (this) {
-            // ignore the state change if we are stopping
-            if (stopping)
+            // Ignore the state change if we are stopping.
```

Comment:
Yes, it's to minimize the impact of potential bugs. It will technically run the risk of masking them since they'll surface as WARN-level logs instead of failed tasks in the REST API; I think it's better to take the less drastic approach here since users are probably going to get scared if they see failed tasks.

I was thinking of [this log message](https://github.com/apache/kafka/blob/45ecae6a28fe820eb2698c8a375c83ee15036f5c/connect/runtime/src/main/java/org/apache/kafka/connect/runtime/Worker.java#L407) when considering how to handle this scenario, since it can be emitted spuriously under certain circumstances. Although that's not great, the fact that the only fallout of being in that unexpected state is a warning log message has had some advantages, since it tends not to alarm users (and possibly act as a red herring).